Read More link enhancements

markbaird opened the pull request:
This includes the changes to "read more" links we discussed in #68.

brianhyder commented on include/service/entities/content/article_service_v2.js:
```diff
@@ -145,23 +145,54 @@ module.exports = function(pb) {
+        }
+
+        ArticleServiceV2.super_.prototype.getSingle.apply(this, [options, cb]);
+    };
```
Good catch.

brianhyder commented on include/service/entities/content/article_service_v2.js:
```diff
@@ -145,23 +145,54 @@ module.exports = function(pb) {
+
+    /**
+     *
+     * @method getAll
```
Can you update the name to ```getSingle```

brianhyder commented on include/service/entities/content/article_service_v2.js:
```diff
@@ -145,23 +145,54 @@ module.exports = function(pb) {
-        
         var self = this;
+        var renderOptions = {};
+        if (!options || options.readMore === undefined) {
```
I think you can simplify the logic here by doing:
```
var renderOptions = {
  readMore: options && options.readMore ? true : false
};
```

brianhyder commented on include/service/entities/content/article_service_v2.js:
```diff
@@ -145,23 +145,54 @@ module.exports = function(pb) {
             
             //complete the rendering
-            self.render([article], function(err/*, articles*/) {
+            self.render([article], renderOptions, function(err/*, articles*/) {
```
One small nit-picky item.  Would you mind moving the creation of the ```renderOptions``` block here?  It marries the logic the options to the code that utilizes it.

brianhyder commented on test/include/service/entities/content/article_renderer_tests.js:
```diff
@@ -0,0 +1,68 @@
+//depedencies
+var should          = require('should');
+var UrlService      = require('../../../../../include/service/entities/url_service.js')({config: {siteRoot: 'http://www.test.com'}});
+var ArticleRenderer = require('../../../../../include/service/entities/content/article_renderer.js')({UrlService: UrlService});
```
These tests are awesome!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

brianhyder commented:
@markbaird this is great.  Would you also mind looking at the section, article, and blog controllers to make sure they are passing the new option correctly?

markbaird commented:
@brianhyder I already tested the blog controller and it is working correctly. I'll make the changes you noted and test those controllers later today.

markbaird commented:
@brianhyder also, is Coveralls not working anymore? I wanted to see if the new tests increased the coverage any.

markbaird commented on plugins/pencilblue/controllers/section.js:
```diff
@@ -120,7 +120,7 @@ module.exports = function(pb) {
             };
             pb.ArticleServiceV2.setPublishedClause(opts.where);
-            self.service.getBySectionViewController(section, opts, function(err, content) {
+            self.service.getBySection(section, opts, function(err, content) {
```
This was throwing an exception because getBySectionViewController() doesn't exist.

brianhyder replied:
Fail on my part.  Thank you for correcting.

markbaird commented:
I implemented your suggestions and tested the blog, article and section controllers again. The section controller actually had a typo causing an exception to be thrown. I checked that I hadn't introduced the bug by verifying it was present on master, then I fixed it and included the fix in this pull request.
